src/hotspot/share/gc/shared/pretouchTask.cpp

```diff
@@ -28,59 +28,99 @@
 #include "runtime/atomic.hpp"
 #include "runtime/globals.hpp"
 #include "runtime/os.hpp"
+#include "utilities/ticks.hpp"
 
 PretouchTask::PretouchTask(const char* task_name,
                            char* start_address,
                            char* end_address,
                            size_t page_size,
-                           size_t chunk_size) :
+                           size_t chunk_size,
+                           uint n_threads,
+                           size_t task_status) :
     AbstractGangTask(task_name),
     _cur_addr(start_address),
     _start_addr(start_address),
     _end_addr(end_address),
     _page_size(page_size),
-    _chunk_size(chunk_size) {
+    _chunk_size(chunk_size),
+    _n_threads(n_threads),
+    _task_status(task_status) {
 
   assert(chunk_size >= page_size,
          "Chunk size " SIZE_FORMAT " is smaller than page size " SIZE_FORMAT,
          chunk_size, page_size);
 }
 
+void PretouchTask::reinitialize(char* start_addr, char* end_addr) {
+  Atomic::release_store(&_cur_addr, start_addr);
+  Atomic::release_store(&_end_addr, end_addr);
```

Comment:
Back-to-back release-stores have redundant loadstore semantics. You could just use a storestore() barrier after the first release_store, then use plain Atomic::store for the second field.

Reply:
OK, will change as per your suggestion.